django/template/base.py

```diff
@@ -757,6 +757,11 @@ def __init__(self, var):
             raise TypeError(
                 "Variable must be a string or number, got %s" % type(var))
         try:
+            # Python treats 'inf' as a float. We don't want to treat it as
+            # a literal.
+            if var.lower() == 'inf':
```

Comment:
I believe `'nan'` has the same issue.

```
>>> float('nan')
nan
```

Reply:
From the ticket:

>Using 'nan' as a template variable name works OK:
```python
>>> Template('{{ nan }}').render(Context({'nan': 'xxx'}))
u'xxx'
```

Reply:
Correct, that's because `int(float('nan'))` later raises `ValueError`. However the problem also exists for "Infinity", "-Infinity", and various casings of those values.

Reply:
If it does not already exist, might be good capture the `nan` behavior in a test to help demonstrate existing correctness.